com.unity.render-pipelines.high-definition/Runtime/PostProcessing/PostProcessSystem.cs

```diff
@@ -3454,9 +3468,8 @@ FinalPassParameters PrepareFinalPass(HDCamera hdCamera, BlueNoise blueNoise, boo
             {
                 if (dynResHandler.HardwareDynamicResIsEnabled())
                 {
-                    var scaledSize = dynResHandler.GetLastScaledSize();
-                    backBufferRect.width = scaledSize.x;
-                    backBufferRect.height = scaledSize.y;
+                    backBufferRect.width = destinationHandle.rt.width;
```

Comment:
This is not always correct. This is used only when PostProcessIsFinalPass is false, that means that the texture is created using our own system.
The HW dynamic res doesn't necessarily need to render to the final resolution of the destination target. Imagine this scenario: 

- Camera 0 (e.g. main camera): has a final resolution of 4k, Dynamic Res at 50%. Then this code is fine as it sets to the desired final res. 
- Camera 1 (e.g. realtime reflection probe): has a final resolution of 512x512. Destination now is 4k as that is the biggest resolution found (given how RTHandle works), then this code will set the viewport to the full 4k, having then wrong UVs calculated. 

The code that was there before looks wrong to me (maybe I am forgetting something @JulienIgnace-Unity  ?), but a solution I think could be just getting the output from final resolution (queriable from the dyn res handler) instead of the buffer size.

Reply:
I tried this with realtime reflection probes. The reason why it works there is because DRS is disabled. This code path only gets executed if DRS is on, DRS is hardware mode, and we have a pass after FinalPass (blit into some UI). The reason for this from what I understand is to reduce complexity by using the entire resolution of the target. I could remove this altogether though, and achieve the same results. Any reason why we'd want to keep this code? Maybe @JulienIgnace-Unity  can answer?

Reply:
> probes. The reason why it works there is because DRS is disabled

Yes, maybe my example was bad, but same thing can happen with any secondary camera that uses dyn res too (as long we have dyn res on two things with different target final resolutions we'll have problems)

Reply:
Actually, @FrancescoC-unity  this is completely unnecesary once I fixed the side effect issue below :)

Reply:
Ok, using finalViewport works like a charm. But wouldnt this be equivalent of deleting this entire block of code? (the backbuffer rect is set to the finalViewport from the camera, which is not the DRS viewport, is the proper final viewport).
It gets set on line 572 of HDCamera.cs
            DynamicResolutionHandler.instance.finalViewport = new Vector2Int((int)finalViewport.width, (int)finalViewport.height);

Reply:
I am indeed a bit confused on why that code is there to start with :/ Waiting on the Julien summon he might know why.

Reply:
ok, on the meantime slapped a TODO there. Hopefully we can nuke this whole block and simplify things a bit :)

Reply:
As far as I can tell, the viewport size should indeed always be finalViewport here (as we want anything rendered after post processes at full rez. Debug modes mostly). I don't think it should be destinationHandle.rt.width though as it would not work with 2 cameras using dynamic rez and different output resolution).
The other important thing here is setting x,y to 0,0 since all subsequent debug assumes viewports starts at (0,0)

Reply:
Let's nuke those lines then!

Reply:
💪